### DISCUSSION TITLE
zoomToElements Decorator Issue

### DISCUSSION THREAD

**Discussion by g3rezz** [iTwin/community#19] [Q&A] [answered]
Hello, I am trying to zoom to a `Decorator` in the viewport. The `Decorator` is a `GraphicType.Scene` with a `viewport.iModel.transientIds.getNext()`. 

Despite the `Decorator` getting highlighted, there's no zooming effect upon button click. I suspect it might be related to the `Decorator` lacking a bounding box. Is this the case? What's the correct approach to achieve zooming into a `Decorator`?

Relevant code:

```tsx
const ZoomToElementsWidget = () => {
	const iModelConnection = useActiveIModelConnection();
	
	const applyZoom = async () => {
	    if (iModelConnection) {
	      const id = "0xffffff0000000001"; // Decorator transientId
	      const vp = IModelApp.viewManager.selectedView;
	      if (vp) {
	        await vp.zoomToElements(id, {
	          animateFrustumChange: true,
	          standardViewId: StandardViewId.Iso,
	        });
	      }
	      iModelConnection.selectionSet.replace(id);
	    }
	  };
	
	  return (
	    <div className="sample-options">
	      <div className="sample-grid">
	        <Button
	          styleType="high-visibility"
	          className="zoom-controls-d"
	          onClick={applyZoom}
        >	
	          Zoom to Selected Element
	        </Button>
	      </div>
	    </div>
	  );
	};
```

The target here is a `Decorator`, not a persistent element within the iModel.
Any insights on resolving this scenario would be highly appreciated. Thank you!

**Comment by pmconne** [comment, marked as answer]
`zoomToElements` queries the backend for the bounding boxes of the elements specified. It can't do anything useful with decorations because they're not elements.

Once it has the bounding boxes ("placements"), it forwards them to [zoomToPlacements](https://www.itwinjs.org/reference/core-frontend/views/viewport/zoomtoplacements/).

If you have or can compute a [Placement](https://www.itwinjs.org/reference/core-common/geometry/placement/) for your decoration, pass it to `zoomToPlacements` directly.
